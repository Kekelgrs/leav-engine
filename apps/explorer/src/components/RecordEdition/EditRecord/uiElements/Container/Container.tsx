// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {Col, Row} from 'antd';
import React from 'react';
import {useFormElementsByContainerContext} from '../../hooks/useFormElementsByContainerContext';
import {IFormElementProps} from '../../_types';

function Container({element, record, recordValues}: IFormElementProps<{}>): JSX.Element {
    const formElements = useFormElementsByContainerContext();
    const children = formElements[element.id] ?? [];

    return (
        <>
            {children.map(el => (
                <Row data-testid="container-child-element" key={el.id}>
                    <Col span={24}>
                        {el.uiElement && <el.uiElement element={el} record={record} recordValues={recordValues} />}
                    </Col>
                </Row>
            ))}
        </>
    );
}

export default Container;
