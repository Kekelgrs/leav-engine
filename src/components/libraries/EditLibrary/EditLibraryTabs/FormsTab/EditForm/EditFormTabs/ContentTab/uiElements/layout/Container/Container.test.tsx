import {render} from 'enzyme';
import React from 'react';
import {DndProvider} from 'react-dnd';
import {TestBackend} from 'react-dnd-test-backend';
import Container from './Container';

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect
}));

describe('Container', () => {
    test('Snapshot test', async () => {
        const comp = render(
            <DndProvider backend={TestBackend}>
                <Container />
            </DndProvider>
        );

        expect(comp).toMatchSnapshot();
    });
});
