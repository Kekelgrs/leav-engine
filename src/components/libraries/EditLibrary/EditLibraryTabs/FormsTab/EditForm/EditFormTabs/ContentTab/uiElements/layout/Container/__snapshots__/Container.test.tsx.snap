// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Snapshot test 1`] = `
initialize {
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
